hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -1107,10 +1107,12 @@ def index_selects_values_in_order(index):
 @pytest.mark.parametrize(
     "condition",
     [
-        lambda ix: Ellipsis in ix,
-        lambda ix: Ellipsis not in ix,
-        lambda ix: np.newaxis in ix,
-        lambda ix: np.newaxis not in ix,
+        lambda ix: isinstance(ix, tuple) and Ellipsis in ix,
+        lambda ix: isinstance(ix, tuple) and Ellipsis not in ix,
+        lambda ix: isinstance(ix, tuple) and np.newaxis in ix,
+        lambda ix: isinstance(ix, tuple) and np.newaxis not in ix,
+        lambda ix: ix is Ellipsis,
+        lambda ix: ix == np.newaxis,
     ],
 )
```

Comment:
this test only passed before by a stroke of fate: it finds the example it needs and bails early, before it hits a case where it tries to iterate over a non-tuple.

This doesn't get hit on master, but it will get hit on #3818, so I'm fixing it early here.